src/github_runner_manager/reactive/consumer.py

```diff
@@ -92,37 +117,46 @@ def consume(
 
     Raises:
         JobError: If the job details are invalid.
+        QueueError: If an error when communicating with the queue occurs.
     """
-    with Connection(queue_config.mongodb_uri) as conn:
-        with closing(SimpleQueue(conn, queue_config.queue_name)) as simple_queue:
-            with signal_handler(signal.SIGTERM):
-                msg = simple_queue.get(block=True)
-                try:
-                    job_details = cast(JobDetails, JobDetails.parse_raw(msg.payload))
-                except ValidationError as exc:
-                    msg.reject(requeue=True)
-                    raise JobError(f"Invalid job details: {msg.payload}") from exc
-                logger.info(
-                    "Received job with labels %s and job_url %s",
-                    job_details.labels,
-                    job_details.url,
-                )
-                if not _validate_labels(
-                    labels=job_details.labels, supported_labels=supported_labels
-                ):
-                    logger.error(
-                        "Found unsupported job labels in %s. "
-                        "Will not spawn a runner and requeue the message.",
+    try:
+        with Connection(queue_config.mongodb_uri) as conn:
+            with closing(SimpleQueue(conn, queue_config.queue_name)) as simple_queue:
+                with signal_handler(signal.SIGTERM):
```

Comment:
Nitpick: (no need to re-impl)
May I suggest using multiple `with`s to reduce nesting?
```suggestion
with Connection(queue_config.mongodb_uri) as conn, closing(SimpleQueue(conn, queue_config.queue_name)) as simple_queue, with signal_handler(signal.SIGTERM):
```